src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const incomingData = [...this];
```

Comment:
Just curios: why did you create a copy instead of working with the original array?

Reply:
Yep, it's unnecessary here. You don't mutate `incomingData` array, so no need to make a copy of `this`.